File: 2016_10_30/comb_sort/comb_sort.js
#!/usr/bin/node

function comb(a) {
	let gap = a.length;
	let swaps = true;
	while (gap > 1 || swaps) {
		gap = Math.max(1, Math.floor(gap / 1.3));
		swaps = false;
		for (let i = 0; i < a.length - gap; i++) {
			let j = i + gap;
			if (a[i] > a[j]) {
				swaps = true;
				[a[i], a[j]] = [a[j], a[i]];
			}
		}
	}
}
let size = Number.parseInt(process.argv[2]);
let test = [];
for (let i = 0; i < size; i++)
	test.push(i);

for (let i = 0; i < size; i++) {
	let j = Math.floor(Math.random() * (size - i)) + i;
	[test[i], test[j]] = [test[j], test[i]];
}

console.log(test);
comb(test);
console.log(test);

File: 2016_10_30/comb_sort/comb_sort.lua
#!/usr/bin/lua

function comb(a)
	local c = 0
	local gap = #a
	local swaps = true
	while gap > 1 or swaps do
		gap = math.max(1, math.floor(gap / 1.3))
		swaps = false
		for i = 1, #a - gap do
			c = c + 1
			j = i + gap
			if a[i] > a[j] then
				a[i], a[j] = a[j], a[i]
				swaps = true
			end
		end
	end
	print("Comparisons: ", c)
end

local size = tonumber(arg[1])
local test = {}
math.randomseed(os.time())
for i = 1, size do test[i] = math.random(0,size) end
print(table.concat(test, " "))
comb(test)
print(table.concat(test, " "))
